src/redis-cli.c

```diff
@@ -1796,6 +1800,22 @@ static int cliSendCommand(int argc, char **argv, long repeat) {
        works well with the interval option. */
     while(repeat < 0 || repeat-- > 0) {
         redisAppendCommandArgv(context,argc,(const char**)argv,argvlen);
+
+        if (is_unsubscribe_command) {
+            /* In unsubscribe related commands, we need to read the specified
+             * number of replies according to the number of parameters. */
+            argc--; /* Skip the command */
+            do {
+                if (cliReadReply(output_raw) != REDIS_OK) {
+                    cliPrintContextError();
+                    exit(1);
+                }
+                fflush(stdout);
+            } while(--argc);
+            zfree(argvlen);
+            return REDIS_OK;
```

Comment:
It seems that this `return` bypasses the surrounding `while(repeat < 0 || repeat-- > 0)` loop. I think it should be `continue` instead of `return`.

The purpose of the loop is to allow repeateing a command N times, e.g. `2 SUBSCRIBE x y` means send the command twice.

Reply:
> this was actually read the reply twice.

Right, the added loop is for reading multiple replies. I was referring to the outer while loop, which sends the command multiple times.

I think this fix is good. In my PR I added a loop in exactly the same place for reading the appropriate number of replies for [s|p][un]subscribe. The same loop is also reading any pubsub messages while waiting for a reply for another command. There'll be a merge conflict for sure but it will be easy anyway.